src/main/java/com/cloudbees/plugins/credentials/impl/CertificateCredentialsImpl.java

```diff
@@ -156,6 +158,21 @@ private static char[] toCharArray(@NonNull Secret password) {
         return password.getPlainText().toCharArray();
     }
 
+    /**
+     * When serializing over a {@link Channel} ensure that we send a self-contained version.
+     *
+     * @return the object instance to write to the stream.
+     */
+    private Object writeReplace() {
```

Comment:
this is wrong.
You calling code should just call the snapshot taker to get a snapshot before using it.